I18n date formats

bjthompson805 opened the pull request:
Many dates were using a `DateFormat` module and are now using `i18n` instead.

### What github issue is this PR for, if any?
Resolves #1408

### What changed, and why?
Any place that was using the `DateFormat` module to format dates has been changed to use the `I18n.l` function instead.

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪
I've written a test in `spec/decorators/casa_case_decorator_spec.rb` which checks for the correct date format of a case's updated_at time after it has been updated.

### Screenshots please :)
Here is an example of where a date has been switched from the `DateFormat` module to `i18n` and is working properly to display the date -- the text "Admin was deactivated on: 12-19-2020" shows this.
![image](https://user-images.githubusercontent.com/40772561/102696902-07df5c80-41ef-11eb-9004-25636db8069f.png)

### Feelings gif (optional)
![alt text](https://media.giphy.com/media/c0wo4QbREPSvRAnNIF/giphy.gif)

bjthompson805 committed:
Removes the DateFormat module

bjthompson805 committed:
Moves all date formats to i18n.

Many dates were using a DateFormat module and are now using i18n instead.

littleforest commented on config/locales/en.yml:
```diff
@@ -52,3 +52,8 @@ en:
   time:
     formats:
       day_and_date: "%A, %v"
+      mm_dd_yyyy: "%m-%d-%Y"
```
Can we give this one an actual name, instead of one that describes the date format? While this app will probably not need to be localized, in the spirt of leaving that option open, we should use names that make sense in other languages. For example, if we were to deploy this in Europe, this format would probably look like:

```
de:
  time:
    formats:
      mm_dd_yyyy: "%d-%m-%Y"
```

in which case the label is less useful.  We could use something like "standard" or "default" for the label, or anything else that makes sense to you.

littleforest reviewed (changes requested):
@bjthompson805 thanks so much for working on this! Can you also fix the merge conflict?

bjthompson805 committed:
Merge branch 'main' into i18n-date-formats

bjthompson805 committed:
Renames "mm_dd_yyyy" to "standard" and removes a use of DateFormat module.

Due to other countries using dd-mm-yyyy format, changed it to the name
"standard" instead. There was also a use of the DateFormat module still,
due to merge with main, so that's been fixed as well.

compwron commented:
This is looking very promising! There are some test failures. they look like

```
      ActionView::Template::Error:
        uninitialized constant CaseContactDecorator::DateFormat
        Did you mean?  DateAndTime
```
Maybe a merge issue?

bjthompson805 commented:
Yes I've fixed that one (a merged file needed to be migrated to i18n). I am having some other failures after merging as well that I can't figure out. I posted in the Slack casa channel.

littleforest reviewed (approved):
🎉 💯 🚢 Thanks for working on this!
